src/black/lines.py

```diff
@@ -623,9 +619,23 @@ def _maybe_empty_lines(self, current_line: Line) -> Tuple[int, int]:
         user_had_newline = bool(before)
         depth = current_line.depth
 
+        # Mutate self.previous_defs, remainder of this function should be pure
         previous_def = None
         while self.previous_defs and self.previous_defs[-1].depth >= depth:
             previous_def = self.previous_defs.pop()
+        if current_line.is_decorator or current_line.is_def or current_line.is_class:
+            if not current_line.is_decorator:
+                self.previous_defs.append(current_line)
```

Comment:
```suggestion
        if current_line.is_def or current_line.is_class:
            self.previous_defs.append(current_line)
```
This should be equivalent unless I'm missing something

Reply:
Thanks, fixed. Logic was previously inside `_maybe_empty_lines_for_class_or_def` and I'd just copied the guard around that function call.